duckduckgo/iOS
# Updates to Autofill from Android ship review

Task/Issue URL: https://app.asana.com/0/0/1204241917356223/f
Tech Design URL:
CC:

**Description**:
Applying changes to Autofill as required from the Android ship review



**Steps to test this PR**:

Note: translations are in progress

***Surpress fireproof dialog when prompted to save Autofill login***
1. Ensure fireproof prompts are enable as well as autofill Logins.
2. Save a new valid login for a site e.g. amazon, netflix etc
3. Confirm the prompt to save a login is presented and that there is no prompt to fireproof
4. Update the login for a site while logging in
5. Confirm the prompt to update the login passwrod is presented and that there is no prompt to fireproof
6. Login to a site using an existing login from the autofill prompt
7. Confirm the prompt to fireproof is presented

***Order of logins in suggested section***
1. Manually create the following logins:
a. title=News, website=nytimes.com, username=anything, password=anything
b. title=Entertainment, website=nytimes.com, username=anything, password=anything
c. title=LEAVE EMPTY, website=nytimes.com, username=anything, password=anything

2. Visit nytimes.com and then access autofill from the overflow menu
3. In the suggested section, confirm the sites are presented in this order:
1st = Entertainment
2nd = News
3rd = nytimes.com

***Additional logins for site***
1. Confirm the same message is shown to the user when creating a new login for a site and when creating an additional login for the same site

***Copy updates - Complete list of changes is [here](https://app.asana.com/0/72649045549333/1204245049760679)***

1. Starting with no Logins saved:
2. Confirm the autofill logins option is now called `"Logins"` from:
a. Settings screen
b. Browsing menu
c. Title of logins list screen

2. Disable device passcode
3. Confirm the Logins screen title = `"Secure your device to save Logins"` and body = `"A passcode is required to protect your Logins."`

4. Reenable device passcode
5. Confirm the unlock Autofill screen now only shows the padlock icon with no text
6. Confirm the passcode screen now says `"Unlock device to access saved Logins"`

7. On the Logins screen confirm the title in the main content of the screen is always `"No Logins saved yet"`, regardless of Autofill switch state

8. Save a new login in browser
9. Confirm prompt title is `"Use a saved Login?"`

10. Update the username of a saved login
11. Confirm prompt title is now `"Update username?"` and the button title is `"Update Username"`

12. Update the password of a saved login
13. Confirm prompt title is now `"Update password for <username>?"` and the button title is `"Update Password"`

14. In the Logins list screen confirm the search bar placeholder text is `"Search logins"`

15. Trigger alert to keep using autofill
16. Confirm alert title = `"Do you want to keep saving Logins?"`, message is `"You can disable this at any time in Settings"` and button titles are `"Disable"` and `"Keep Saving"`

**Autofill Login prompt always contained "From This Website", reported [here](https://app.asana.com/0/0/1204321489594248/f)**
1. On a site with only 1 login that is an exact match, confirm the autofill prompt does *not* contain the label `From This Website`
![image](https://user-images.githubusercontent.com/91189922/230416833-863f6efa-1a53-4c5f-982e-855de524db89.png)

2. Add a login for the same site, add a subdomain to the url so it is no longer an exact match e.g. `sub.fill.dev`

3. Confirm the autofill prompt now contains the label `From This Website` above exact match login, and that, after tapping More Options, the label `From sub.fill.dev` is above the partially matching login
![image](https://user-images.githubusercontent.com/91189922/230417295-f129d572-4741-4113-96f7-6c4c9022060a.png)

4. Delete the exact match login and confirm the label From sub.fill.dev is above the partially matching login
![image](https://user-images.githubusercontent.com/91189922/230416962-39d12a00-8bd7-49a9-a6e1-63709b6a2782.png)




**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
